File: tut/6thu/whales.pl
#!/usr/bin/perl -w

$curr_word = "";
$curr_count = 0;

while ($line = <STDIN>) {
    chomp $line;
    if ($line =~ /([0-9]+) (.*)/) {
        $word = $2;
        $count = $1;

        if ($word eq $curr_word) {
            $curr_count += $count;
        } else {
            if ($curr_count > 0) {
                printf("%d %s\n", $curr_count, $curr_word);
            }
            $curr_count = $count;
            $curr_word = $word;
        }
    }
}
if ($curr_count > 0) {
    printf("%d %s\n", $curr_count, $curr_word);
}

File: tut/6thu/whales.py
#!/usr/bin/python2.7 -u

import sys
import re

curr_word = ""
curr_count = 0

for line in sys.stdin:
  m = re.match("([0-9]+) (.*)", line)
  if m:
    word = m.group(2)
    count = int(m.group(1))
    
    if word == curr_word:
      curr_count += count
    else:
      if curr_count > 0:
        print "%d %s" % (curr_count, curr_word)
      curr_count = count
      curr_word = word

if curr_count > 0:
  print "%d %s" % (curr_count, curr_word)
